### PR TITLE
convert StorageDirectoryParcelable to Kotlin

### PR DESCRIPTION
**Pull request by shalva97**
#### Issue tracker   
Wanted to contribute something simple, so here it is...

#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [x] Done  
  

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

**Comment by shalva97** [comment]
@NanyangTaiji You are crazy, Java is deprecated

**Comment by VishnuSanal** [comment]
This PR looks okay. Since this is just refractoring, it can be merged once all tests pass. We'll look into why the tests fail.

**Comment by TranceLove** [comment]
Agreed with @VishnuSanal - allow me to find a Linux box to run the PR tonight for possible problems.

However @shalva97 no, Java is here to stay, though perhaps not so much in Android landscape. I'm happy to see Java being pushed forward as JVM languages emerge and evolve as well.

Disclaimer: love to write Kotlin over Java too

**Review by TranceLove — APPROVED** [review]
It should be good to go. Local build succeeded; test failure at Github Actions was caused by something unrelated to this PR, just I've yet to find a way to retry CI without new commits...

**Review by VishnuSanal — APPROVED** [review]
tests passes now. the CI fails at publishing test results, which we'll look into.

File: app/src/main/java/com/amaze/filemanager/adapters/data/StorageDirectoryParcelable.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+package com.amaze.filemanager.adapters.data
+
+import android.os.Parcel
+import android.os.Parcelable
+import androidx.annotation.DrawableRes
+
+/** Identifies a mounted volume  */
+data class StorageDirectoryParcelable(
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/adapters/data/StorageDirectoryParcelable.kt:28]: Isn't data class inherently parcelable ?

**Pull request merged**